Apps/Examples/Examples/All Examples/DebugMapExample.swift

```diff
@@ -0,0 +1,179 @@
+        map.debugOptions = controller.enabledDebugOptions
+    }
+}
+
+private final class SettingsViewController: UIViewController, UITableViewDataSource {
+
+    weak var delegate: DebugOptionSettingsDelegate?
+    private var listView: UITableView!
+
+    private(set) var enabledDebugOptions: [MapDebugOptions]
+    private let allSettings: [MapDebugOptionSetting] = [
+        MapDebugOptionSetting(debugOption: .collision, displayTitle: "Debug collision"),
+        MapDebugOptionSetting(debugOption: .depthBuffer, displayTitle: "Show depth buffer"),
+        MapDebugOptionSetting(debugOption: .overdraw, displayTitle: "Debug overdraw"),
+        MapDebugOptionSetting(debugOption: .parseStatus, displayTitle: "Show tile coordinate"),
+        MapDebugOptionSetting(debugOption: .renderCache, displayTitle: "Render Cache"),
+        MapDebugOptionSetting(debugOption: .stencilClip, displayTitle: "Show stencil buffer"),
+        MapDebugOptionSetting(debugOption: .tileBorders, displayTitle: "Debug tile clipping"),
+        MapDebugOptionSetting(debugOption: .timestamps, displayTitle: "Show tile loaded time"),
+    ]
+
+    init(debugOptions: [MapDebugOptions]) {
+        enabledDebugOptions = debugOptions
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        listView = UITableView()
+        listView.dataSource = self
+
+        view.addSubview(listView)
+
+        listView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            listView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            listView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor),
+            listView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            listView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor),
+        ])
+
+        navigationItem.largeTitleDisplayMode = .never
+        navigationItem.rightBarButtonItem = UIBarButtonItem(
+            barButtonSystemItem: .save,
+            target: self,
+            action: #selector(saveSettings(_:)))
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        preferredContentSize = listView.contentSize
+    }
+
+    @objc private func saveSettings(_ sender: UIBarButtonItem) {
+        delegate?.debugOptionSettingsDidChange(self)
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        allSettings.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cellID = String(describing: DebugOptionCell.self)
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellID) as? DebugOptionCell ?? DebugOptionCell()
```

Comment:
I recommend registering the cell class for the cell reuse identifier and then using [`dequeueReusableCell(withIdentifier:for:)`](https://developer.apple.com/documentation/uikit/uitableview/1614878-dequeuereusablecell) which returns a non-optional cell. You'll still need a cast, but I think a force-cast would be fine since you control all the inputs.